Jenkinsfile

```diff
@@ -86,14 +86,23 @@ node('docker && imgtec') {  // Only run on internal slaves as build takes a lot
             }
 
             // Build all (for opkg)
-            // TODO grab vault creds and mod config to use OPKGSMIME
             if (params.ALL_PACKAGES){
                 echo 'Enabling all user and kernel packages'
                 sh 'echo \'' \
                  + 'CONFIG_ALL=y\n' \
                  + '\' >> .config'
             }
 
+            // Add opkg signing key
+            withCredentials([
+                [$class: 'FileBinding', credentialsId: 'opkg-build-private-key', variable: 'PRIVATE_KEY'],
+                [$class: 'FileBinding', credentialsId: 'opkg-build-public-key', variable: 'PUBLIC_KEY'],
+            ]){
+                echo 'Copying private and public key pair'
+                sh "cp ${env.PRIVATE_KEY} ${WORKSPACE}/opkg-build-key && \
+                    cp ${env.PUBLIC_KEY} ${WORKSPACE}/opkg-build-key.pub"
```

Comment:
This is a security breach! The credentials are hidden for a reason, now you are just copying them to a known location and not cleaning up which will leave the keys lying around on the slaves. The whole point of the 'with' block is that it cleans resources automatically so that the credentials are not available except for inside the block.

To get around your private/pubic key naming issue I suggest something like this:
sh "ln -s ${env.PUBLIC_KEY}  ${env.PRIVATE_KEY}.pub"

Then you need to put any make commands inside this block, I suggest you just move the block to wrap everything inside the build stage.